File: backend/models/vendor.go
package models

import (
	"gorm.io/gorm"
)

type Vendor struct {
	gorm.Model
	Name          string        `json:"name" gorm:"not null"`
	ContactPerson string        `json:"contact_person"`
	Phone         string        `json:"phone"`
	Email         string        `json:"email"`
	Address       string        `json:"address"`
	Description   string        `json:"description"`
	JobID         *uint         `json:"job_id"`
	Job           *Job          `json:"job" gorm:"foreignKey:JobID"`
	PaymentTerms  []PaymentTerm `json:"payment_terms" gorm:"foreignKey:VendorID"`
}

type PaymentTerm struct {
	gorm.Model
	VendorID    uint      `json:"vendor_id" gorm:"not null;index"`
	TermNumber  int       `json:"term_number" gorm:"not null"`
	Percentage  float64   `json:"percentage"`
	Amount      float64   `json:"amount"`
	DueDate     string    `json:"due_date"`
	Description string    `json:"description"`
}

func (v *Vendor) AfterFind(tx *gorm.DB) error {
	if v.Job != nil && v.Job.ID == 0 {
		v.Job = nil
	}
	return nil
}

File: mobile/lib/models/vendor.dart
class Vendor {
  final int id;
  final String name;
  final String? contactPerson;
  final String? phone;
  final String? email;
  final String? address;
  final String? description;
  final int? jobId;
  final Map<String, dynamic>? job;

  Vendor({
    required this.id,
    required this.name,
    this.contactPerson,
    this.phone,
    this.email,
    this.address,
    this.description,
    this.jobId,
    this.job,
  });

  factory Vendor.fromJson(Map<String, dynamic> json) {
    return Vendor(
      id: json['ID'] ?? json['id'] ?? 0,
      name: json['Name'] ?? json['name'] ?? '',
      contactPerson: json['ContactPerson'] ?? json['contact_person'],
      phone: json['Phone'] ?? json['phone'],
      email: json['Email'] ?? json['email'],
      address: json['Address'] ?? json['address'],
      description: json['Description'] ?? json['description'],
      jobId: json['JobID'] ?? json['job_id'],
      job: json['Job'] ?? json['job'],
    );
  }
}

class PaymentTerm {
  final int id;
  final int vendorId;
  final int termNumber;
  final double? percentage;
  final double? amount;
  final String? dueDate;
  final String? description;

  PaymentTerm({
    required this.id,
    required this.vendorId,
    required this.termNumber,
    this.percentage,
    this.amount,
    this.dueDate,
    this.description,
  });

  factory PaymentTerm.fromJson(Map<String, dynamic> json) {
    return PaymentTerm(
      id: json['ID'] ?? json['id'] ?? 0,
      vendorId: json['VendorID'] ?? json['vendor_id'] ?? 0,
      termNumber: json['TermNumber'] ?? json['term_number'] ?? 0,
      percentage: (json['Percentage'] ?? json['percentage'])?.toDouble(),
      amount: (json['Amount'] ?? json['amount'])?.toDouble(),
      dueDate: json['DueDate'] ?? json['due_date'],
      description: json['Description'] ?? json['description'],
    );
  }
}
